clang/test/SemaHLSL/resource_binding_attr_error.hlsl

```diff
@@ -44,7 +44,7 @@ void foo2() {
   // expected-warning@+1 {{'register' attribute only applies to cbuffer/tbuffer and external global variables}}
   extern RWBuffer<float> U2 : register(u5);
 }
-// FIXME: expect-error once fix https://github.com/llvm/llvm-project/issues/57886.
+// expected-error@+1 {{invalid register name prefix 'u' for 'float' (expected 't')}}
```

Comment:
Why a float expected 't'?

Reply:
I don't have a prior example to base this off of at hand, but I think I saw some hlsl with a builtin type bound to a register and the valid prefix was 't'.
To answer your question, the assignment is represented as a VarDecl, so the compiler assumes this resource is a Sampler, UAV, or  SRV. Then the resource type is analyzed to see if it's a builtin (float happens to be a builtin).
Then we check the binding prefix to see if it's 't', and if not, emit an error. float is a builtin type, and the prefix isn't t, so that's why this test case expects 't'.
What prefix should it expect, if any?

Reply:
dxc will report
error: invalid register specification, expected 'b', 'c', or 'i' binding
float t : register(u0);
          ^
fxc will just ignore the register(u0).

Reply:
Since here the float is in global scope, it will be treated as constant inside Global CBuffer.

Reply:
According to @tex3d 's comment, the ` expected 'b', 'c', or 'i' binding` is no longer supported, so I don't think I should weave that into this implementation. I did ask what we should expect the prefix to be, but I haven't received an answer yet. Sounds to me like you think it should be 'b', because it's a constant inside a CBuffer?

Reply:
We should consider deprecating the register binding of this style: `float b : register(c0);`, since it's not supported by DXC, and was only supported by FXC for DX9 targets.  Register binding should only be applicable to global resource/sampler declarations and cbuffer/tbuffer declarations.  So the answer to "what prefix should that be" is: we don't support register bindings on values that go into the constant buffer, so there is no valid prefix to use here.  The closest equivalent to these legacy register bindings that we do support are `packoffset` annotations.